tests/btest/web-socket/timestamp.py

```diff
@@ -0,0 +1,286 @@
+# @TEST-GROUP: web-socket
+#
+# @TEST-PORT: BROKER_WEB_SOCKET_PORT
+#
+# @TEST-EXEC: btest-bg-run node "broker-node --config-file=../node.cfg"
+# @TEST-EXEC: btest-bg-run recv "python3 ../recv.py >recv.out"
+# @TEST-EXEC: $SCRIPTS/wait-for-file recv/ready 15 || (btest-bg-wait -k 1 && false)
+#
+# @TEST-EXEC: btest-bg-run send "python3 ../send.py"
+#
+# @TEST-EXEC: $SCRIPTS/wait-for-file recv/done 30 || (btest-bg-wait -k 1 && false)
+# @TEST-EXEC: btest-diff recv/recv.out
+#
+# @TEST-EXEC: btest-bg-wait -k 1
+
+@TEST-START-FILE node.cfg
+
+broker {
+  disable-ssl = true
+}
+topics = ["/test"]
+verbose = true
+
+@TEST-END-FILE
+
+@TEST-START-FILE recv.py
+
+import asyncio, websockets, os, time, json, sys, re
+
+
+from datetime import datetime
+
+ws_port = os.environ['BROKER_WEB_SOCKET_PORT'].split('/')[0]
+
+ws_url = f'ws://localhost:{ws_port}/v1/messages/json'
+
+# Expected timestamps in the message.
+timestamps = [
+    # using UTC (note: 'Z' is not supported on older Python versions)
+    '2020-02-02T02:02:02+00:00',
+    '2020-02-02T02:02:02.1+00:00',
+    '2020-02-02T02:02:02.12+00:00',
+    '2020-02-02T02:02:02.123+00:00',
+    '2020-02-02T02:02:02.1234+00:00',
+    '2020-02-02T02:02:02.12345+00:00',
+    '2020-02-02T02:02:02.123456+00:00',
+    # using positive offset from UTC
+    '2020-02-02T02:02:02+02:00',
+    '2020-02-02T02:02:02.1+02:00',
+    '2020-02-02T02:02:02.12+02:00',
+    '2020-02-02T02:02:02.123+02:00',
+    '2020-02-02T02:02:02.1234+02:00',
+    '2020-02-02T02:02:02.12345+02:00',
+    '2020-02-02T02:02:02.123456+02:00',
+    # using negative offset from UTC
+    '2020-02-02T02:02:02-02:00',
+    '2020-02-02T02:02:02.1-02:00',
+    '2020-02-02T02:02:02.12-02:00',
+    '2020-02-02T02:02:02.123-02:00',
+    '2020-02-02T02:02:02.1234-02:00',
+    '2020-02-02T02:02:02.12345-02:00',
+    '2020-02-02T02:02:02.123456-02:00',
+]
+
+# tells btest we're done by writing a file
+def write_done_file():
+    with open('done', 'w') as f:
+        f.write('done')
+
+def parse_iso_timestamp(timestamp):
+    # replace 'Z' and adjust offset format from '[+-]MM:SS' to '[+-]MMSS'
+    # (note: Python versions >= 3.7 don't need this)
```

Comment:
Following up on our earlier conversation Dominik: you can safely assume 3.7+. We bumped our minimum requirement a while back, see [here](https://docs.zeek.org/en/master/install.html#required-dependencies).

I'm about to submit a round of CI updates since Broker's setup has fallen behind compared to Zeek `master` and I believe at least the (EOL'd) Ubuntu 18.04 that's currently still configured ran Python 3.6.

Reply:
Unfortunately it's not that easy: https://github.com/zeek/zeek/issues/3239